subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/ResourceVersionLister.java

```diff
@@ -54,7 +54,7 @@ public void listVersions(ModuleIdentifier module, IvyArtifactName artifact, List
         for (ResourcePattern pattern : patterns) {
             visit(pattern, artifact, module, collector, result);
         }
-        // TODO:DAZ Be a bit smarter about this
+        // Could be a bit smarter about this
```

Comment:
These kinds of comments are unhelpful, as they don't tell me what's wrong and how to fix it. Instead there should be a method level comment saying what the current approach is and what its drawbacks are.

Reply:
You're absolutely right. I don't remember why I put that there in the first place. Will investigate and either remove the comment or make it more useful.